indexer/database/bridge_transfers.go

```diff
@@ -127,31 +126,41 @@ type L1BridgeDepositsResponse struct {
 	HasNextPage bool
 }
 
-// L1BridgeDepositsByAddress retrieves a list of deposits intiated by the specified address, coupled with the L1/L2 transaction
-// hashes that complete the bridge transaction.
+// L1BridgeDepositsByAddress retrieves a list of deposits intiated by the specified address,
+// coupled with the L1/L2 transaction hashes that complete the bridge transaction.
 func (db *bridgeTransfersDB) L1BridgeDepositsByAddress(address common.Address, cursor string, limit int) (*L1BridgeDepositsResponse, error) {
 	defaultLimit := 100
 	if limit <= 0 {
 		limit = defaultLimit
 	}
 
-	depositsQuery := db.gorm.Table("l1_bridge_deposits").Select(`
-l1_bridge_deposits.*,
-l1_contract_events.transaction_hash AS l1_transaction_hash,
-l1_transaction_deposits.l2_transaction_hash`)
-
 	// TODO join with l1_tokens and l2_tokens
-	depositsQuery = depositsQuery.Joins("INNER JOIN l1_transaction_deposits ON l1_bridge_deposits.transaction_source_hash = l1_transaction_deposits.source_hash")
-	depositsQuery = depositsQuery.Joins("INNER JOIN l1_contract_events ON l1_transaction_deposits.initiated_l1_event_guid = l1_contract_events.guid")
-
-	if cursor != "" {
-		depositsQuery = depositsQuery.Where("l1_bridge_deposits.transaction_source_hash < ?", cursor)
-	}
-
-	filteredQuery := depositsQuery.Where(&Transaction{FromAddress: address}).Order("l1_bridge_deposits.transaction_source_hash DESC").Limit(limit + 1)
-
+	ethAddressString := predeploys.LegacyERC20ETHAddr.String()
+
+	// Coalesce l1 transaction deposits that are simply ETH sends
+	ethTransactionDeposits := db.gorm.Model(&L1TransactionDeposit{})
+	ethTransactionDeposits = ethTransactionDeposits.Where(Transaction{FromAddress: address}).Where(`data = '0x' AND amount > 0`)
+	ethTransactionDeposits = ethTransactionDeposits.Joins("INNER JOIN l1_contract_events ON l1_contract_events.guid = initiated_l1_event_guid")
+	ethTransactionDeposits = ethTransactionDeposits.Select(`
+from_address, to_address, amount, data, source_hash AS transaction_source_hash,
+l2_transaction_hash, l1_contract_events.transaction_hash AS l1_transaction_hash,
+l1_transaction_deposits.timestamp, NULL AS cross_domain_message_hash, ? AS local_token_address, ? AS remote_token_address`, ethAddressString, ethAddressString)
+
+	depositsQuery := db.gorm.Model(&L1BridgeDeposit{})
+	depositsQuery = depositsQuery.Joins("INNER JOIN l1_transaction_deposits ON l1_transaction_deposits.source_hash = transaction_source_hash")
+	depositsQuery = depositsQuery.Joins("INNER JOIN l1_contract_events ON l1_contract_events.guid = l1_transaction_deposits.initiated_l1_event_guid")
+	depositsQuery = depositsQuery.Select(`
+l1_bridge_deposits.from_address, l1_bridge_deposits.to_address, l1_bridge_deposits.amount, l1_bridge_deposits.data, transaction_source_hash,
+l2_transaction_hash, l1_contract_events.transaction_hash AS l1_transaction_hash,
+l1_bridge_deposits.timestamp, cross_domain_message_hash, local_token_address, remote_token_address`)
```

Comment:
The code you provided seems to be part of a larger program or script, and it appears to involve querying a database using an Object-Relational Mapping (ORM) library called GORM, which is commonly used with Go (Golang) programming language to interact with databases. This code seems to be constructing and executing a series of database queries to retrieve specific information related to transactions and deposits. Let's break down the code step by step:

1. Retrieving ETH Transaction Deposits:

go
Copy code
ethTransactionDeposits := db.gorm.Model(&L1TransactionDeposit{})
ethTransactionDeposits = ethTransactionDeposits.Where(Transaction{FromAddress: address}).Where(`data = '0x' AND amount > 0`)
ethTransactionDeposits = ethTransactionDeposits.Joins("INNER JOIN l1_contract_events ON l1_contract_events.guid = initiated_l1_event_guid")
ethTransactionDeposits = ethTransactionDeposits.Select(`
from_address, to_address, amount, data, source_hash AS transaction_source_hash,
l2_transaction_hash, l1_contract_events.transaction_hash AS l1_transaction_hash,
l1_transaction_deposits.timestamp, NULL AS cross_domain_message_hash, ? AS local_token_address, ? AS remote_token_address`, ethAddressString, ethAddressString)
This sequence of statements performs the following actions:

It initializes a query object ethTransactionDeposits for the L1TransactionDeposit table.
It applies filtering conditions to the query: deposits with a specific FromAddress, where data is '0x' and amount is greater than zero.
It performs an inner join with the l1_contract_events table using the initiated_l1_event_guid relationship.
It selects specific columns from the tables, using aliases for some columns. The ethAddressString variable is used as a placeholder for local_token_address and remote_token_address.
2. Retrieving L1 Bridge Deposits:

go
Copy code
depositsQuery := db.gorm.Model(&L1BridgeDeposit{})
depositsQuery = depositsQuery.Joins("INNER JOIN l1_transaction_deposits ON l1_transaction_deposits.source_hash = transaction_source_hash")
depositsQuery = depositsQuery.Joins("INNER JOIN l1_contract_events ON l1_contract_events.guid = l1_transaction_deposits.initiated_l1_event_guid")
depositsQuery = depositsQuery.Select(`
l1_bridge_deposits.from_address, l1_bridge_deposits.to_address, l1_bridge_deposits.amount, l1_bridge_deposits.data, transaction_source_hash,
l2_transaction_hash, l1_contract_events.transaction_hash AS l1_transaction_hash,
l1_bridge_deposits.timestamp, cross_domain_message_hash, local_token_address, remote_token_address`)
This series of statements performs similar actions for L1 Bridge Deposits:

Initializes a query object depositsQuery for the L1BridgeDeposit table.
Performs inner joins with the l1_transaction_deposits and l1_contract_events tables, establishing relationships.
Selects specific columns from the tables, using aliases as needed.
These two sections of code suggest that the program is retrieving deposit-related information from a database, potentially to process or analyze it further. It's important to note that this interpretation is based on the context you provided, and the actual behavior and purpose of the code could vary based on the larger context of the program it's a part of.